common/src/lib.rs

```diff
@@ -1,8 +1,14 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
-    }
+pub mod value;
+
+pub use value::Value;
+
+/// This enum is equivalent to a completion record in cases where only normal
+/// and throw completions are allowed – except that it also allows
+/// [`Self::Killed`] as a variant to allow termination of a running script.
```

Comment:
> "Allow" termination? This seems.. interesting?
> When would we want to allow it, but not demand it?


Well it would be handy to have for uncaught errors or promise rejections

Reply:
I didn't mean "allow but not demand". I meant that we want killing a running script (maybe after some timeout, or after a signal from another thread) to be an option, and that would need a separate "type of completion" (in spec language) than normal completions or exceptions.

The JS spec doesn't contemplate scripts being killed, but the HTML spec does: https://html.spec.whatwg.org/multipage/webappapis.html#killing-scripts